### PR TITLE
chore: dark mode

### LINKED ISSUE #78: Support dark mode ✨
Since we use Chakra UI for the docs. It'll be great to add dark mode support to the docs

### PR DESCRIPTION
**Pull request by jer3m01**
Continuation of #85.

All done! Ready for review.

Feedback highly appreciated! 

Closes #78
Closes #85

### CODE REVIEW

**Comment by segunadebayo** [comment]
Great work here @jer3m01.

I prefer to use semantic tokens instead of `useBreakpointValue` or `_dark` pseudo prop. 

Let's create semantic tokens in the format:

(bg|text|border)-(primary|secondary|default)-(subtle|bold)

CleanShot 2023-01-18 at 12 50 41@2x

File: components/icons.tsx [hunk 1]
```diff
@@ -110,14 +111,14 @@ export function StatechartIcon(props: Props) {
         cx="21.9539"
         cy="21.9539"
         r="10.2039"
-        stroke="black"
+        stroke={useColorModeValue("black", "white")}
```
Comment (anubra266) [inline components/icons.tsx:114]: We don't want to use `useColorModeValue`. What you can do is set `stroke` to `currentColor`, then set the color as a semantic token on the `svg` itself.

More explanation;
You can change the parent `svg` to `<chakra.svg />`
Set the color that can change there
```js
 <chakra.svg
   width="94"
   height="94"
   viewBox="0 0 94 94"
   fill="none"
   /* This is just for explanation, use a semantic token instead */
   color="black"
   _dark={{
      color: "white",
   }}
   {...props}
>
```

Then in your `circle`
```js
 <circle
   stroke="currentColor" 
/>
```

Comment (jer3m01) [reply components/icons.tsx]: Sadly the `{...props}` passed to the svg element do not work on chakra.svg and cause a typescript error (appears in one of the Vercel failed deployments logs).

Instead I gave it an `id` and added a css style.

**Comment by jer3m01** [comment]
Converted it to semantic tokens (except for certain unique cases as I don't think they require their own token).

File: components/workshop-link.tsx [hunk 2]
```diff
@@ -15,6 +15,10 @@ export function WorkshopLink() {
       align="center"
       href="/creator-workshop"
       className="group"
+      _dark={{
+        bg: "orange.900",
+        color: "orange.300",
+      }}
```
Comment (anubra266) [inline components/workshop-link.tsx:21]: We can use semanticTokens here

File: components/top-navigation.tsx [hunk 3]
```diff
@@ -38,12 +40,18 @@ export function TopNavigation() {
             fontSize="xs"
             letterSpacing="wider"
             fontWeight="semibold"
+            display={{ base: "none", sm: "block" }}
+            borderBottomColor="border-default-subtle"
+            _dark={{
+              bg: "orange.900",
+              color: "orange.300",
+            }}
```
Comment (anubra266) [inline components/top-navigation.tsx:48]: We can use semantic tokens here

File: layouts/docs.tsx [hunk 4]
```diff
@@ -46,10 +46,16 @@ export default function DocsLayout({ children, doc, toc }: DocsLayoutProps) {
             overscrollBehavior="contain"
           >
             <Box position="relative">
-              <Box position="sticky" top="0" bg="white" pb="8">
-                <Spacer height="10" bg="white" />
+              <Box
+                position="sticky"
+                top="0"
+                bg="white"
+                _dark={{ bg: "gray.800" }}
```
Comment (anubra266) [inline layouts/docs.tsx:53]: We can use semantic tokens here

File: theme/global-styles.ts [hunk 5]
```diff
@@ -37,6 +40,9 @@ export const globalStyles: SystemStyleObject = {
     marginY: "8",
     th: {
       bg: "gray.100",
+      _dark: {
+        bg: "gray.700",
+      },
```
Comment (anubra266) [inline theme/global-styles.ts:45]: We can use semantic tokens here

File: theme/layer-styles.ts [hunk 6]
```diff
@@ -25,6 +27,9 @@ const inlineCode: SystemStyleObject = {
   fontFamily: "mono",
   fontWeight: "semibold",
   color: "pink.600",
+  _dark: {
+    bg: "whiteAlpha.100",
+  },
```
Comment (anubra266) [inline theme/layer-styles.ts:32]: We can use semantic tokens here

**Pull request merged**